### PR TITLE
MailGroup & MailGroupMessage

### PR DESCRIPTION
**Pull request by romo-odoo**

### CODE REVIEW

File: addons/mail_group/static/src/interactions/mail_group_message.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    onClickLinkShow() {
+        const container = ev.currentTarget.closest(".o_mg_link_parent");
+        container.find('.o_mg_link_hide').first().classList.remove('d-none');
+        container.find('.o_mg_link_show').first().classList.add('d-none');
+        container.find('.o_mg_link_content').first().classList.add('d-none');
+    }
+
+    onClickReadMore() {
+        rpc(ev.target.getAttribute('href'), {
```
Comment (ged-odoo) [inline addons/mail_group/static/src/interactions/mail_group_message.js]: async function, then you can use await, and also, wrap it in a waitFor

File: addons/mail_group/static/src/interactions/mail_group_message.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        quoted.insertBefore(readMore);
+        readMore.addEventListener("click", () => quoted.classList.toggle("visible"));
+    }
+
+    onClickLinkHide() {
+        const container = ev.currentTarget.closest(".o_mg_link_parent");
+        container.querySelector('.o_mg_link_hide').classList.add('d-none');
+        container.querySelector('.o_mg_link_show').classList.remove('d-none');
+        container.querySelector('.o_mg_link_content').classList.remove('d-none');
+    }
```
Comment (robinlej) [inline addons/mail_group/static/src/interactions/mail_group_message.js]: same as the other PR: toggle a boolean and use `t-att-class`

File: addons/mail_group/static/src/interactions/mail_group_message.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            }
+            const threadContainer = ev.target.closest(".o_mg_replies")?.querySelector("ul.list-unstyled");
+            if (threadContainer) {
+                let lastMsg = threadContainer.querySelectorAll("li.media").last;
+                const newMessages = data.querySelector("ul.list-unstyled").querySelectorAll(":scope > li.media");
+                for (const newMessage in newMessages) {
+                    lastMsg.insertAdjacentElement(newMessage, "afterend");
+                    lastMsg = newMessage;
+                }
+                threadContainer.appendChild(data.querySelector('.o_mg_read_more').parentElement)
```
Comment (robinlej) [inline addons/mail_group/static/src/interactions/mail_group_message.js]: `this.insert`

File: addons/mail_group/static/src/interactions/mail_group_message.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    onClickReadMore() {
+        rpc(ev.target.getAttribute('href'), {
+            last_displayed_id: ev.target.dataste.listDisplayedId,
+        }).then(function (data) {
+            if (!data) {
+                return;
+            }
+            const threadContainer = ev.target.closest(".o_mg_replies")?.querySelector("ul.list-unstyled");
+            if (threadContainer) {
+                let lastMsg = threadContainer.querySelectorAll("li.media").last;
```
Comment (robinlej) [inline addons/mail_group/static/src/interactions/mail_group_message.js]: `.last`?

File: addons/mail_group/static/src/interactions/mail_group_message.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                const newMessages = data.querySelector("ul.list-unstyled").querySelectorAll(":scope > li.media");
+                for (const newMessage in newMessages) {
+                    lastMsg.insertAdjacentElement(newMessage, "afterend");
```
Comment (robinlej) [inline addons/mail_group/static/src/interactions/mail_group_message.js]: `this.insert`

File: addons/mail_group/static/src/interactions/mail_group.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            'token': this.token,
+        }));
+
+        this.el.querySelector(".o_mg_alert").remove();
+
+        if (response === 'added') {
+            this.isMember = true;
+            const btnEl = this.el.querySelector(".o_mg_subscribe_btn");
+            btnEl.innerText = _t('Unsubscribe');
+            btnEl.classList.remove("btn-primary");
```
Comment (robinlej) [inline addons/mail_group/static/src/interactions/mail_group.js]: Classes... so many classes... Don't want to use `t-att-class`? :eyes:

File: addons/mail_group/static/src/interactions/mail_group_message.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        const quoted = this.el.querySelector(".card-body *[data-o-mail-quote]");
+        const readMore = document.createElement("button");
+        readMore.classList.add("btn btn-light btn-sm ms-1");
+        readMore.innerText = ". . .";
+        quoted.insertBefore(readMore);
+        readMore.addEventListener("click", () => quoted.classList.toggle("visible"));
+    }
+
+    async onClickReadMore() {
+        await this.waitFor(rpc(ev.target.getAttribute('href'), {
```
Comment (ged-odoo) [inline addons/mail_group/static/src/interactions/mail_group_message.js]:
```js
const data = await this.waitFor(...);
```

**Pull request merged**